src/SystemInterface.py

```diff
@@ -51,6 +51,120 @@ def check_systemctl_status_for_component(
+
+            # Call the readline is blocking, set it to non-blocking mode.
+            os.set_blocking(process.stdout.fileno(), False)
+
+            while True:
+                # Check timeout
+                if time.time() > timeout:
+                    print(f"Timeout after {timeout} seconds")
+                    process.terminate()
+                    return False
+
+                output, errors = process.communicate()
+                if output:
+                    print(output)
+                    print(errors)
+                    return True
+
+                # Check if process has terminated
+                if process.poll() is not None:
+                    print("Shutdown process terminated.")
+                    return False
+
+                time.sleep(0.01)
+
+        except KeyboardInterrupt:
+            print("\nExiting the shutdown attempt...")
+            process.terminate()
+            return False
+        except Exception as e:
+            print(f"Error: {e}")
+            return False
+        finally:
+            # Ensure process is terminated
+            try:
+                process.terminate()
+            except:
+                pass
+
+    def start_systemd_nucleus_lite(self, timeout: int | float) -> bool:
+        try:
+            cmd = [
+                "sudo",
+                "systemctl",
+                "start",
+                "--with-dependencies",
+                "greengrass-lite.target",
+            ]
+
+            # Run the command and stream output
+            process = subprocess.Popen(
+                cmd,
+                stdout=subprocess.PIPE,
+                stderr=subprocess.PIPE,
+                universal_newlines=True,
+            )
+
+            timeout = time.time() + timeout
+
+            # Call the readline is blocking, set it to non-blocking mode.
+            os.set_blocking(process.stdout.fileno(), False)
+
+            while True:
+                # Check timeout
+                if time.time() > timeout:
+                    print(f"Timeout after {timeout} seconds")
+                    process.terminate()
+                    return False
+
+                output, errors = process.communicate()
+                if output:
+                    print(output)
+                    print(errors)
+                    return False
+
+                # Check if process has terminated
+                if process.poll() is not None:
+                    print("Shutdown process terminated.")
+                    return False
+
+                time.sleep(0.01)
+
+        except KeyboardInterrupt:
+            print("\nExiting the shutdown attempt...")
+            process.terminate()
+            return False
+        except Exception as e:
+            print(f"Error: {e}")
+            return False
+        finally:
+            # Ensure process is terminated
+            try:
+                process.terminate()
+            except:
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

It appears that you are generically passing an `Exception` object without performing any other operation on it. This may hide error conditions that can otherwise be quickly detected and addressed. We recommend that you catch a more specific exception. If the code must broadly catch all exceptions, consider logging the stack trace using the [logging.exception()](https://docs.python.org/3/library/logging.html#logging.exception) API. For example,

```
try:
    x = 1 / 0
except ZeroDivisionError as e:
    logging.exception(&#39;ZeroDivisionError: %s&#39;, e)

```